statsmodels/base/model.py

```diff
@@ -436,14 +436,17 @@ def fit(self, start_params=None, method='newton', maxiter=100,
         nobs = self.endog.shape[0]
         f = lambda params, *args: -self.loglike(params, *args) / nobs
         score = lambda params, *args: -self.score(params, *args) / nobs
-        try:
+        if observed_hessian:
             hess = lambda params, *args: -self.hessian(params, *args) / nobs
-        except:
-            hess = None
+        else:
+            hess = lambda params, *args: -self.hessian(params, *args, observed=False) / nobs
```

Comment:
nice trick
I was wondering why this doesn't break things in the standard case where observed is not an available keyword.
But you don't change the code on the default path.

I don't know why we had the try with catch-all except.